R/endpoints.R

```diff
@@ -118,6 +117,41 @@ endpoint_validate_input <- function(req, res, type, path) {
   hintr_response(response, "ValidateInputResponse")
 }
 
+#' Validate survey and programme data including consistency with shape file
+#' returning an indication of success and if successful return the data
+#' required by UI.
+#'
+#' @param req The request as PlumberRequest object.
+#' @param res The response as a PlumberResponse object.
+#' @param type The type of file to validate: ANC, survey or programme.
+#' @param path Path to the file to validate.
+#' @param shape Path to shape file for comparison.
+#'
+#' @return Validated JSON response with data and incidcation of success.
+#' @keywords internal
+endpoint_validate_survey_programme <- function(req, res, type, path, shape) {
+  validate_json_schema(req$postBody, "ValidateSurveyAndProgrammeRequest")
+  validate_func <- switch(type,
+                          programme = do_validate_programme,
+                          anc = do_validate_anc,
+                          survey = do_validate_survey)
+  validate_if_exists <- function(path, shape) {
```

Comment:
why does this function shadow the package-level function? is there a simpler way of containing this logic?

Reply:
Fixed that now to just check for file existing within the `with_success` call.